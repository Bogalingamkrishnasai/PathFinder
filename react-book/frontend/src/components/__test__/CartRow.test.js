import { render, screen, fireEvent } from '@testing-library/react';
import CartRow from '../CartRow';

describe('CartRow Component', () => {
  const mockCartItem = { itemId: '1', quantity: 2 };
  const mockItems = [{ itemId: '1', title: 'Test Item', price: 10, salePrice: 8 }];
  const mockDispatch = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders cart item details', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <CartRow cartItem={mockCartItem} dispatch={mockDispatch} items={mockItems} />
        </tbody>
      </table>
    );
    
    expect(screen.getByText('Test Item')).toBeInTheDocument();
    expect(screen.getByText('2')).toBeInTheDocument();
    expect(screen.getByText('$16.00')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('calls dispatch when remove button is clicked', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <CartRow cartItem={mockCartItem} dispatch={mockDispatch} items={mockItems} />
        </tbody>
      </table>
    );
    
    fireEvent.click(screen.getByText('X'));
    expect(mockDispatch).toHaveBeenCalledWith({
      type: 'REMOVE',
      itemId: '1'
    });
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders correctly with sale price', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <CartRow cartItem={mockCartItem} dispatch={mockDispatch} items={mockItems} />
        </tbody>
      </table>
    );
    
    expect(screen.getByText('$16.00')).toBeInTheDocument(); // 2 × $8 sale price
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders correctly with regular price when no sale price', () => {
    const itemsWithoutSale = [{ ...mockItems[0], salePrice: undefined }];
    const { asFragment } = render(
      <table>
        <tbody>
          <CartRow cartItem={mockCartItem} dispatch={mockDispatch} items={itemsWithoutSale} />
        </tbody>
      </table>
    );
    
    expect(screen.getByText('$20.00')).toBeInTheDocument(); // 2 × $10 regular price
    expect(asFragment()).toMatchSnapshot();
  });
});