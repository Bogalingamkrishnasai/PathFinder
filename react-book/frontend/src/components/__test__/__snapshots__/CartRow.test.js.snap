// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartRow Component calls dispatch when remove button is clicked 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td>
          2
        </td>
        <td>
          Test Item
        </td>
        <td>
          $16.00
        </td>
        <td>
          <button
            type="button"
          >
            X
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`CartRow Component renders cart item details 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td>
          2
        </td>
        <td>
          Test Item
        </td>
        <td>
          $16.00
        </td>
        <td>
          <button
            type="button"
          >
            X
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`CartRow Component renders correctly with regular price when no sale price 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td>
          2
        </td>
        <td>
          Test Item
        </td>
        <td>
          $20.00
        </td>
        <td>
          <button
            type="button"
          >
            X
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`CartRow Component renders correctly with sale price 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td>
          2
        </td>
        <td>
          Test Item
        </td>
        <td>
          $16.00
        </td>
        <td>
          <button
            type="button"
          >
            X
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
